spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestCallStatementParser.java

```diff
@@ -137,6 +139,28 @@ public void testCallParseError() {
         () -> parser.parsePlan("CALL cat.system radish kebab"));
   }
 
+  @Test
+  public void testCallStripsLeadingComments() throws ParseException {
+    // These comments are meant to look like those that systems like DBT would add to statements.
+    List<String> callStatementsWithComments = Lists.newArrayList(
```

Comment:
Can you please in in tests for comments interspersed and after valid commands?

Reply:
Can you explain that one more time? You're asking for tests with comments inside of and after valid commands? I can add that.

I'm also going to add a test that actually calls the `CALL` procedure with a comment in it.

Reply:
I added tests for comments interspersed with valid commands.

I also added an end to end test, although it's only testing one command at present. The normal `spark.sql` parser that we fall back to doesn't seem to handle comments very well if there are two or more statements in it.

Reply:
So I think this will solve the user's problem, as only one statement is executed at a time in dbt if I'm not mistaken (and this fix is needed regardless to recognize Iceberg commands with comments in them.

But for multiple statements being processed, we'd have to add some splitting around `;` like the Spark Hive CLI does https://github.com/apache/spark/commit/450418bdfdc46c2569ff22c307de15c4fca76ebc

Reply:
In fact, plain `spark.sql` can't handle multiple commands at once.

I tried this in a test, `sql("INSERT INTO TABLE %s VALUES (1, 'a'); INSERT INTO TABLE %s VALUES(2, 'a');", tableName, tableName)`, and got the following error message:

```
== SQL ==
INSERT INTO TABLE testhive.default.table VALUES (1, 'a'); INSERT INTO TABLE testhive.default.table VALUES(2, 'a');
----------------------------------------------------------^^^

	at app//org.apache.spark.sql.catalyst.parser.ParseException.withCommand(ParseDriver.scala:266)
	at app//org.apache.spark.sql.catalyst.parser.AbstractSqlParser.parse(ParseDriver.scala:127)
	at app//org.apache.spark.sql.execution.SparkSqlParser.parse(SparkSqlParser.scala:51)
	at app//org.apache.spark.sql.catalyst.parser.AbstractSqlParser.parsePlan(ParseDriver.scala:77)
	at app//org.apache.spark.sql.catalyst.parser.extensions.IcebergSparkSqlExtensionsParser.parsePlan(IcebergSparkSqlExtensionsParser.scala:123)
```

So that's falling back to the delegate parser (Spark's own parser). But the case of a single SQL command having comments at the beginning (or in the middle or at the end) is covered by this change. Previously, it would have worked for commands that are not covered by `isIcebergCommend` - e.g. anything starting with `call` or some subset of `alter table` commands.

Reply:
The full error message is this 

```
extraneous input 'INSERT' expecting {<EOF>, ';'}(line 1, pos 58)

== SQL ==
INSERT INTO TABLE testhive.default.table VALUES (1, 'a'); INSERT INTO TABLE testhive.default.table VALUES(2, 'a');
----------------------------------------------------------^^^
```

Reply:
Anyway, the tests have been added 🙂